clang/include/clang/Sema/Sema.h

```diff
@@ -1158,7 +1158,9 @@ class Sema final {
       if (FD) {
         FD->setWillHaveBody(true);
         S.ExprEvalContexts.back().InImmediateFunctionContext =
-            FD->isImmediateFunction();
+            FD->isImmediateFunction() ||
+            S.ExprEvalContexts[S.ExprEvalContexts.size() - 2]
```

Comment:
Someday... I would like to replace this with a function call so I can stop going "wait, is -2 okay??" each time I see it. ;-)